notebooks/registry.ipynb

```diff
@@ -0,0 +1,615 @@
+        "\n",
+        "For example, let’s consider the following scenario: Your `agent` class has an argument `arg1` which is annotated to be `List[Class1]` ( where `Class1` is `Registrable`), and the `Class1` constructor takes an argument `arg2`. In the passed YAML config, there are two different `Class1` object configs listed. The constructor will check to see if both `agent.arg1.0.arg2` and `agent.arg1.1.arg2` have been passed.\n",
+        "\n",
+        "The parameters passed in the command line will be parsed according to the type annotation of the corresponding low level constructor. If it is not one of `int`, `float`, `str`, or `bool`, it simply loads the string into python using a YAML loader.\n",
+        "\n",
+        "Each constructor returns the object, as well a dictionary config with all the parameters used to create the object and any `Registrable` objects created in the process of creating this object."
+      ],
+      "metadata": {
+        "id": "Cxo_vGT4-vJW"
+      }
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "### Registering an Environment\n",
+        "\n",
+        "Consider registering a custom environment class named `Grid` (which inherits `BaseEnv`) in the RLHive registry. "
+      ],
+      "metadata": {
+        "id": "YrlRz676D_Dm"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "from hive.envs.base import BaseEnv\n",
+        "class Grid(BaseEnv):\n",
+        "    def __init__(self, env_name = 'Grid', **kwargs):\n",
+        "        pass\n",
+        "    def reset(self):\n",
+        "        pass\n",
+        "    def step(self):\n",
+        "        pass\n",
+        "    def render(self):\n",
+        "        pass\n",
+        "    def close(self):\n",
+        "        pass\n",
+        "    def save(self):\n",
+        "        pass"
+      ],
+      "metadata": {
+        "id": "PbefnmFLJprJ"
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "from hive.utils.registry import registry\n",
+        "registry.register(name = 'Grid', \n",
+        "                  constructor = Grid, \n",
+        "                  type = BaseEnv)"
+      ],
+      "metadata": {
+        "id": "J3DGzZgP-u28"
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "More than one environment can be registered at once using the `register_all` method. Consider registering three environments, `Gridv1`, `Gridv2`, and `Gridv3` in the RLHive registry."
+      ],
+      "metadata": {
+        "id": "sOy_VsfPLAIp"
+      }
+    },
```

Comment:
I think a short description about the arguments of the register_all method would be helpful here.